### DISCUSSION TITLE
How to validate SMILES?

### DISCUSSION THREAD

**Discussion by Koongoon** [rdkit/rdkit#7677] [Q&A] [answered]
When I want to determine if SMILES is chemically correct compound, I wonder if there are some methods for this.

There will be valence tests, binding tests, etc., and I wonder if there is a method implemented for this.

In addition, MolFromSmiles is known as a method that converts smiles into chem.mol. I wonder how much validation it goes through.

**Comment by bp-kelley** [comment, marked as answer]
The full list of sanitization checks is in the Molecule Sanitization chapter of the RDKit book:

 https://www.rdkit.org/docs/RDKit_Book.html#

in general, the RDKit is very opinionated about molecules and tries to reject nonsensical molecules.  This only goes so far, you are free to make radicals and charge states that can’t really exist.  You can also completely turn off sanitization and roll your own.
